contracts/test/GrantRound.test.ts

```diff
@@ -0,0 +1,153 @@
+
+      await roundContract.connect(mpUser).acceptMatchingPool(balances[0]);
+      expect(await roundContract.connect(deployer).checkBalance(roundContract.address)).to.be.equal(
+        ethers.utils.parseEther(balances[0])
+      );
+    });
+  });
+
+  describe('donateToGrant - Add funds during active round', () => {
+    it('sends donation token to the grant payee with a given grant id', async function () {
+      const grantId: number = 0;
+      await mockERC20.mock.balanceOf.withArgs(donor.address).returns(ethers.utils.parseEther(balances[0]));
+      await mockERC20.mock.balanceOf.withArgs(grantPayee.address).returns(ethers.utils.parseEther(balances[0]));
+      await mockERC20.mock.transfer.withArgs(grantPayee.address, balances[0]).returns(true);
+
+      await roundContract.connect(donor).donateToGrant(balances[0], grantId);
+      expect(await roundContract.connect(deployer).checkBalance(grantPayee.address)).to.be.equal(
+        ethers.utils.parseEther(balances[0])
+      );
+    });
+
+    it('donations revert if not above minimum contribution', async function () {
+      const grantId: number = 0;
+      await mockERC20.mock.balanceOf.withArgs(donor.address).returns(ethers.utils.parseEther(balances[0]));
+      await mockERC20.mock.balanceOf.withArgs(grantPayee.address).returns(ethers.utils.parseEther(balances[0]));
+      await mockERC20.mock.transfer.withArgs(grantPayee.address, balances[0]).returns(true);
+
+      void expect(roundContract.connect(donor).donateToGrant(10, grantId)).to.be.revertedWith(
+        'Donation amount must be greater than minimum contribution'
+      );
+    });
+  });
+
+  describe('payoutGrant - payout remaining contract balance to a given address', () => {
+    it('reverts if round has not ended', async function () {
+      await mockERC20.mock.balanceOf.withArgs(deployer.address).returns(ethers.utils.parseEther(balances[0]));
+      await mockERC20.mock.balanceOf.withArgs(roundContract.address).returns(ethers.utils.parseEther(balances[0]));
+      await mockERC20.mock.transfer.withArgs(deployer.address, balances[0]).returns(true);
+
+      void expect(roundContract.connect(grantRoundOwner).payoutGrant(deployer.address)).to.be.revertedWith(
+        'Method must be called after the active round has ended'
+      );
+    });
+
+    it('sends remaining matching pool funds to payout address', async function () {
+      await mockERC20.mock.balanceOf.withArgs(grantPayee.address).returns(ethers.utils.parseEther(balances[0]));
+      await mockERC20.mock.balanceOf.withArgs(roundContract.address).returns(ethers.utils.parseEther(balances[0]));
+      await mockERC20.mock.transferFrom.withArgs(roundContract.address, grantPayee.address, balances[0]).returns(true);
+
+      // End the round for all subsequent tests
+      await timeTravel(deployer.provider, endTime + 1);
+      // await roundContract.connect(grantRoundOwner).payoutGrant(grantPayee.address);
```

Comment:
There seems to be an obscure bug with the mock contract that I'm still trying to work through. Doubtful it's the contract implementation but obviously could be wrong on that :grimacing:

Reply:
Just generically, when pushing I'd recommend leaving this uncommented out so the test fails, that way it's more clear that there's still work to do on this PR

Overall, my impression here is that Waffle mocks aren't the best fit for these tests / testing actual token transfers, and we should probably just replace it with a `MockERC20` contract that behaves like a normal ERC20